### PR TITLE
Style/modal

### PR DESCRIPTION
**Pull request by jackcasstlesjones**
# What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Styling
- [ ] Build
- [ ] Chore

## Description

- Aesthetic changes to text inputs, input modal, needs modal

## Screenshots

Screenshot 2024-12-15 at 10 38 27
Screenshot 2024-12-15 at 10 46 22
Screenshot 2024-12-15 at 10 38 33
Screenshot 2024-12-15 at 10 38 30 (2)
Screenshot 2024-12-15 at 10 38 30

## UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_

- [ ] Semantic HTML implemented?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

## Added/updated tests?

_Please aim to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [x] No, and this is why: just aesthetic changes
- [ ] I need help with writing tests

## Delete branch after merge?

- [x] Yes
- [ ] No

## What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: src/ui/shared/Modal.tsx [hunk 1]
```diff
@@ -52,14 +52,14 @@ export default function Modal({
...
-                    className="px-4 py-2 text-lg font-medium text-white bg-gray-600 rounded-full hover:bg-gray-500 focus:outline-none focus:ring-2 focus:ring-gray-400 w-36 text-nowrap"
+                    className="px-4 py-2 text-lg font-medium text-white bg-gray-600 rounded-full focus:outline-none focus:ring-2 focus:ring-gray-400 w-36 text-nowrap"
                   />
                 )}
                 {forwardButton && (
                   <Button
                     onClick={forwardButton.action}
                     label={forwardButton.label}
-                    className="px-4 py-2 text-lg font-medium text-white bg-twd-primary-purple rounded-full hover:bg-purple-600 focus:outline-none focus:ring-2 focus:ring-purple-400 w-36 text-nowrap"
+                    className="px-4 py-2 text-lg font-medium text-white bg-twd-primary-purple rounded-full focus:outline-none focus:ring-2 focus:ring-purple-400 w-36 text-nowrap"
```
Comment (maxitect) [inline src/ui/shared/Modal.tsx:62]: Another one [anotherone](https://media0.giphy.com/media/3o7WTxyMSVN7lM5I7C/giphy.gif?cid=6c09b952htsv5wzqd6cu2k5wrmvbbqtr2zp0gdvpiwti5nha&ep=v1_internal_gif_by_id&rid=giphy.gif&ct=g)

File: src/ui/shared/Search.tsx [hunk 2]
```diff
@@ -59,7 +59,7 @@ export default function Search({ onSearch, onClear }: SearchProps) {
               value={query}
               onChange={handleInputChange}
               placeholder="Search..."
-              className="w-full bg-twd-background px-3 py-2 rounded-md border border-gray-300"
+              className="w-full bg-twd-background px-3 py-2 rounded-md border border-gray-300 focus:border-none focus:outline-none focus:ring-2 focus:ring-twd-primary-purple"
```
Comment (maxitect) [inline src/ui/shared/Search.tsx:62]: Same same

**Review by maxitect — CHANGES_REQUESTED** [review]
Thanks for another one! Just question around the use of focus: outline-none. See comments

File: src/styles/globals.css [hunk 3]
```diff
@@ -15,5 +15,5 @@ p {
 }
 
 .addToolInput {
-  @apply w-full p-2 rounded bg-transparent text-white border-b border-gray-700;
+  @apply w-full p-2 rounded bg-transparent text-white border-b border-gray-700 focus:outline-none focus:ring-2 focus:ring-twd-primary-purple;
```
[the remark this answers is not in the document]
Comment (jackcasstlesjones) [reply src/styles/globals.css:18]: without it the browser/device overrides the styling, so our purple outline isn't visible. As far as I'm aware, the fact that we're removing it but adding our own custom outline with focus:ring should mean that this element meets accessibility requirements

File: src/ui/shared/Modal.tsx [hunk 4]
```diff
@@ -41,7 +41,7 @@ export default function Modal({
                 <input
                   type="text"
                   placeholder={placeholder}
-                  className="w-9/12 px-3 py-2 mt-4 text-gray-900 rounded-md focus:outline-none focus:ring-2 focus:ring-purple-500"
+                  className="w-9/12 px-3 py-2 mt-4 text-gray-900 rounded-md focus:outline-none focus:ring-2 focus:ring-twd-primary-purple addToolInput"
```
[the remark this answers is not in the document]
Comment (jackcasstlesjones) [reply src/ui/shared/Modal.tsx:44]: without it the browser/device overrides the styling, so our purple outline isn't visible. As far as I'm aware, the fact that we're removing it but adding our own custom outline with focus:ring should mean that this element meets accessibility requirements

File: src/ui/shared/Modal.tsx [hunk 5]
```diff
@@ -52,14 +52,14 @@ export default function Modal({
                   <Button
                     onClick={backButton.action}
                     label={backButton.label}
-                    className="px-4 py-2 text-lg font-medium text-white bg-gray-600 rounded-full hover:bg-gray-500 focus:outline-none focus:ring-2 focus:ring-gray-400 w-36 text-nowrap"
+                    className="px-4 py-2 text-lg font-medium text-white bg-gray-600 rounded-full focus:outline-none focus:ring-2 focus:ring-gray-400 w-36 text-nowrap"
```
[the remark this answers is not in the document]
Comment (jackcasstlesjones) [reply src/ui/shared/Modal.tsx:55]: without it the browser/device overrides the styling, so our purple outline isn't visible. As far as I'm aware, the fact that we're removing it but adding our own custom outline with focus:ring should mean that this element meets accessibility requirements

File: src/ui/shared/Modal.tsx (hunk 1 above)
Comment (jackcasstlesjones) [reply src/ui/shared/Modal.tsx:62]: without it the browser/device overrides the styling, so our purple outline isn't visible. As far as I'm aware, the fact that we're removing it but adding our own custom outline with focus:ring should mean that this element meets accessibility requirements

File: src/ui/shared/Search.tsx (hunk 2 above)
Comment (jackcasstlesjones) [reply src/ui/shared/Search.tsx:62]: without it the browser/device overrides the styling, so our purple outline isn't visible. As far as I'm aware, the fact that we're removing it but adding our own custom outline with focus:ring should mean that this element meets accessibility requirements

**Review by maxitect — APPROVED** [review]
Happy with that, thanks for explaining

**Pull request merged**